avail-subxt/examples/vector_send_msg.rs

```diff
@@ -0,0 +1,182 @@
+use avail_core::{
+	data_proof::{tx_uid, AddressedMessage, BoundedData, Message, SubTrie},
+	AppId, Keccak256,
+};
+use avail_subxt::{
+	api, avail_client::RpcMethods, rpc::KateRpcClient as _, tx, AccountId, AvailClient, Opts,
+};
+
+use binary_merkle_tree::verify_proof;
+use derive_more::Constructor;
+use futures::stream::{FuturesOrdered, TryStreamExt as _};
+use hex_literal::hex;
+use std::collections::HashSet;
+use structopt::StructOpt;
+use subxt::{backend::BlockRef, error::RpcError, utils::H256, Error};
+use subxt_signer::sr25519::dev;
+
+const DATA: &[u8] = b"Test 42";
+const DOMAIN: u32 = 2;
+
+fn messages() -> Vec<Message> {
+	let asset_id = hex!("4554480000000000000000000000000000000000000000000000000000000000");
+
+	vec![
+		Message::ArbitraryMessage(BoundedData::truncate_from(DATA.to_vec())),
+		Message::FungibleToken {
+			asset_id: H256(asset_id),
+			amount: 42_000_000_000_000_000_000u128,
+		},
+	]
+}
+
+/// Send messages and wait until they are in the same block.
+/// It tries up to 5 times to archive the same block.
```

Comment:
what do you mean by _archive the same block_ here?

Reply:
It ensures that generated messages from `messages()` are part of the same block.
It will facilitate the rest of test because you just need to download one unique block.

We need to check that because depending on the time (or completion of the block), we could have one message in block `i` and the other in block `i+1`.